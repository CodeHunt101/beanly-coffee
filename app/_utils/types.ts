export enum Path {
  HOME = "/",
  ABOUT = "/about",
  CREATE_PLAN = "/create-plan",
  NOT_READY = "#",
}

export enum BreakPoints {
  SM = 576,
  MD = 768,
  LG = 992,
  XL = 1200,
  XXL = 1440,
}

export type Styles = {
  readonly [key: string]: string;
};

type BrewMethod = "Capsule" | "Filter" | "Espresso" | null;
type BeanType = "Single Origin" | "Decaf" | "Blended" | null;
type Size = "250g" | "500g" | "1000g" | null;
type GrindOption = "Wholebean" | "Filter" | "Cafeti√©re" | null;
type DeliveryFrequency = "Every week" | "Every 2 weeks" | "Every month" | null;

export type Option =
  | BrewMethod
  | BeanType
  | Size
  | GrindOption
  | DeliveryFrequency;

export enum Step {
  BrewMethod = "step-1",
  BeanType = "step-2",
  Size = "step-3",
  GrindOption = "step-4",
  DeliveryFrequency = "step-5",
}

export type SelectedOptions = {
  [Step.BrewMethod]: BrewMethod;
  [Step.BeanType]: BeanType;
  [Step.Size]: Size;
  [Step.GrindOption]: GrindOption;
  [Step.DeliveryFrequency]: DeliveryFrequency;
};

export type PlanContextType = {
  selectedOptions: Partial<SelectedOptions>;
  setSelectedOption: (step: Step, option: Option) => void;
};
