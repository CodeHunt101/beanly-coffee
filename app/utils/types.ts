export enum Path {
  HOME = "/",
  NOT_READY = "#",
}

export enum BreakPoints {
  SM = 576,
  MD = 768,
  LG = 992,
  XL = 1200,
  XXL = 1440,
}
