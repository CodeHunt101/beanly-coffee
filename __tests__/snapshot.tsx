/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import Page from "@/app/page";
import { Path } from "@/app/_utils/types";
import About from "@/app/about/page";
import CreatePlan from "@/app/create-plan/page";

it(`renders ${Path.HOME} page unchanged`, () => {
  const { container } = render(<Page />);
  expect(container).toMatchSnapshot();
});

it(`renders ${Path.ABOUT} page unchanged`, () => {
  const { container } = render(<About />);
  expect(container).toMatchSnapshot();
});

it(`renders ${Path.CREATE_PLAN} page unchanged`, () => {
  const { container } = render(<CreatePlan />);
  expect(container).toMatchSnapshot();
});
