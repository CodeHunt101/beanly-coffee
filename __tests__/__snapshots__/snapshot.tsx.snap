// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <main>
    <section
      class="hero text-background"
      id="hero-section"
    >
      <div
        class="content"
      >
        <h1
          class="title ff-serif"
        >
          Great coffee made simple.
        </h1>
        <p
          class="description fw-300"
        >
          Start your mornings with the world's best coffees. Try our expertly curated artisan coffees from our best roasters delivered directly to your door, at your schedule.
        </p>
        <button
          class="btn bg-primary ff-serif text-background"
        >
          Create your plan
        </button>
      </div>
    </section>
    <section
      class="products"
    >
      <header
        class="header"
      >
        <h2
          class="title ff-serif lowercase"
        >
          Our Collection
        </h2>
      </header>
      <ul
        class="productList flex"
      >
        <li
          class="productItem"
        >
          <div
            class="productCard flex"
          >
            <img
              alt="Gran Espresso product"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div>
              <h3
                class="productName ff-serif"
              >
                Gran Espresso
              </h3>
              <p
                class="productDescription"
              >
                Light and flavorful blend with cocoa and black pepper for an intense experience.
              </p>
            </div>
          </div>
        </li>
        <li
          class="productItem"
        >
          <div
            class="productCard flex"
          >
            <img
              alt="Planalto product"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div>
              <h3
                class="productName ff-serif"
              >
                Planalto
              </h3>
              <p
                class="productDescription"
              >
                Brazilian dark roast with rich and velvety body, and hints of fruits and nuts.
              </p>
            </div>
          </div>
        </li>
        <li
          class="productItem"
        >
          <div
            class="productCard flex"
          >
            <img
              alt="Piccollo product"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div>
              <h3
                class="productName ff-serif"
              >
                Piccollo
              </h3>
              <p
                class="productDescription"
              >
                Mild and smooth blend featuring notes of toasted almond and dried cherry.
              </p>
            </div>
          </div>
        </li>
        <li
          class="productItem"
        >
          <div
            class="productCard flex"
          >
            <img
              alt="Danche product"
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div>
              <h3
                class="productName ff-serif"
              >
                Danche
              </h3>
              <p
                class="productDescription"
              >
                Ethiopian hand-harvested blend densely packed with vibrant fruit notes.
              </p>
            </div>
          </div>
        </li>
      </ul>
    </section>
    <section
      class="features"
    >
      <div
        class="wrapper"
      >
        <header
          class="header"
        >
          <h2
            class="title ff-serif"
          >
            Why choose us?
          </h2>
          <p
            class="description"
          >
            A large part of our role is choosing which particular coffees will be featured in our range. This means working closely with the best coffee growers to give you a more impactful experience on every level.
          </p>
        </header>
      </div>
      <ul
        class="featureList"
      >
        <li
          class="featureItem"
        >
          <div
            class="featureCard flex"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="featureContent"
            >
              <h3
                class="featureTitle ff-serif"
              >
                Best quality
              </h3>
              <p
                class="featureDescription"
              >
                Discover an endless variety of the worldâ€™s best artisan coffee from each of our roasters.
              </p>
            </div>
          </div>
        </li>
        <li
          class="featureItem"
        >
          <div
            class="featureCard flex"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="featureContent"
            >
              <h3
                class="featureTitle ff-serif"
              >
                Exclusive benefits
              </h3>
              <p
                class="featureDescription"
              >
                Special offers and swag when you subscribe, including 30% off your first shipment.
              </p>
            </div>
          </div>
        </li>
        <li
          class="featureItem"
        >
          <div
            class="featureCard flex"
          >
            <img
              alt=""
              data-nimg="1"
              decoding="async"
              height="40"
              loading="lazy"
              src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
              srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
              style="color: transparent;"
              width="40"
            />
            <div
              class="featureContent"
            >
              <h3
                class="featureTitle ff-serif"
              >
                Free shipping
              </h3>
              <p
                class="featureDescription"
              >
                We cover the cost and coffee is delivered fast. Peak freshness: guaranteed.
              </p>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </main>
</div>
`;
